wait/host.go

```diff
@@ -584,51 +584,69 @@ func (a *HostAwaitility) WaitForTemplateUpdateRequests(namespace string, count i
 }
 
 // NotificationWaitCriterion checks if a Notification meets the given condition
-type NotificationWaitCriterion func(a *HostAwaitility, mur *toolchainv1alpha1.Notification) bool
+type NotificationWaitCriterion func(a *HostAwaitility, mur toolchainv1alpha1.Notification) bool
 
-// WaitForNotification waits until there is a Notification available with the given name and the optional conditions
-func (a *HostAwaitility) WaitForNotification(name string, criteria ...NotificationWaitCriterion) (*toolchainv1alpha1.Notification, error) {
-	var notification *toolchainv1alpha1.Notification
+// WaitForNotifications waits until there is a Notification available with the given name and the optional conditions
+func (a *HostAwaitility) WaitForNotifications(notificationName, notificationType string, criteria ...NotificationWaitCriterion) ([]*toolchainv1alpha1.Notification, error) {
```

Comment:
The function name and description is confusing. It returns an empty slice if there is no notifications at all. In that case it won't wait for anything.
IMO we should add a new parameter for the number of expected notifications. Something like:
```suggestion
// WaitForNotifications waits until there is an expected number of Notifications available for the provided user and with the notification type and which match the conditions (if provided).
func (a *HostAwaitility) WaitForNotifications(username, numberOfNotifications int, notificationType string, criteria ...NotificationWaitCriterion) ([]*toolchainv1alpha1.Notification, error) {
```

Reply:
Updated comment here: 6857dc3b128382b4fc1feef6eaaa85232b22c933
Updated parameter here: 1ddb1a3c4a441b524ea8e21f03903f6dce26147e